eng/validation/build-arcadewithrepo.ps1

```diff
@@ -0,0 +1,265 @@
+            $response = (Invoke-WebRequest -Uri $uri -Headers $headers -Method Get) | ConvertFrom-Json
+            if(1 -eq $response.count)
+            {
+                $contentArray += $response
+            }
+        }
+        ## If there are no last known good builds in the last $days days, then write a warning. 
+        $contentArray | Foreach-Object {$count += $_.count}
+        if($count -eq 0)
+        {
+            Write-warning "There were no successful builds for this repository in the last ${days} days."
+            Exit
+        }
+        
+        return ($contentArray | Sort-Object { $_.value.finishTime } -descending)[0].value.triggerInfo.'ci.sourceSha'
+    }
+}
+
+function Invoke-AzDOBuild(
+    [string] $azdoOrg, 
+    [string] $azdoProject,
+    [int] $buildDefinitionId,
+    [string] $azdoToken,
+    [string] $branchName,
+    [string] $buildParameters)
+{ 
+    $uri = "https://dev.azure.com/${azdoOrg}/${azdoProject}/_apis/build/builds?api-version=5.1"
+    $base64AuthInfo = [Convert]::ToBase64String([Text.Encoding]::ASCII.GetBytes(":${azdoToken}"))
+    $headers = @{"Authorization"="Basic $base64AuthInfo"}
+
+    $body = @{
+        "definition"=@{
+            "id"=$buildDefinitionId
+        };
+        "sourceBranch"=$branchName;
+    }
+
+    if("" -ne $buildParameters)
+    {
+        $body = $body += @{"parameters"=$buildParameters}
+    }
+
+    $content = Invoke-WebRequest -Uri $uri -Headers $headers -ContentType "application/json" -Body ($body | ConvertTo-Json) -Method Post 
+    return ($content | ConvertFrom-Json).id
+}
+
+function Get-BuildStatus(
+    [string] $azdoOrg, 
+    [string] $azdoProject,
+    [int] $buildId,
+    [string] $azdoToken)
+{
+    $uri = "https://dev.azure.com/${azdoOrg}/${azdoProject}/_apis/build/builds/${buildId}?api-version=5.1"
+    $base64AuthInfo = [Convert]::ToBase64String([Text.Encoding]::ASCII.GetBytes(":${azdoToken}"))
+    $headers = @{"Authorization"="Basic $base64AuthInfo"}
+    $content = Invoke-WebRequest -Uri $uri -Headers $headers -ContentType "application/json" -Method Get 
+    return ($content | ConvertFrom-Json).status
+}
+
+function Get-BuildResult(
+    [string] $azdoOrg, 
+    [string] $azdoProject,
+    [int] $buildId,
+    [string] $azdoToken)
+{
+    $uri = "https://dev.azure.com/${azdoOrg}/${azdoProject}/_apis/build/builds/${buildId}?api-version=5.1"
+    $base64AuthInfo = [Convert]::ToBase64String([Text.Encoding]::ASCII.GetBytes(":${azdoToken}"))
+    $headers = @{"Authorization"="Basic $base64AuthInfo"}
+    $content = Invoke-WebRequest -Uri $uri -Headers $headers -ContentType "application/json" -Method Get 
+    return ($content | ConvertFrom-Json).result
+}
+
+function Get-Github-RepoAuthUri($repoName)
+{
+    "https://${githubUser}:${githubPAT}@github.com/${githubOrg}/${repoName}"
+}
+
+function GitHub-Clone($repoName) 
+{
+    $authUri = Get-Github-RepoAuthUri $repoName
+    & git clone $authUri $(Get-Repo-Location $repoName)
```

Comment:
Should this also use Git-Command?

Reply:
Oops, nope, it's not going to work. Git-Command takes in the repoName and calls Push-Location on it, but since it doesn't exist before the clone, it errors out :(